lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -5351,13 +5351,12 @@ namespace Js
 
     Var JavascriptOperators::NewVarFromDetachedState(DetachedStateBase* state, JavascriptLibrary *library)
     {
-        switch (state->GetTypeId())
+        if (state->GetTypeId() == TypeIds_ArrayBuffer)
         {
-        case TypeIds_SharedArrayBuffer:
-            return Js::SharedArrayBuffer::NewFromSharedState(state, library);
-        case TypeIds_ArrayBuffer:
             return Js::ArrayBuffer::NewFromDetachedState(state, library);
-        default:
+        }
+        else
+        {
             AssertMsg(false, "We should explicitly have a case statement for each object which has detached state.");
```

Comment:
nit: you can do AssertOrFailFastMsg(state->GetTypeId() != TypeIds_ArrayBuffer, "blah blah"); This will be lot cleaner. 

---
In reply to: [115386419](https://github.com/Microsoft/ChakraCore/pull/2939#discussion_r115386419) [](ancestors = 115386419)